File: OC/显示层动画/WGOCAnimationPrimary/WGOCAnimationPrimary/位置&形状&颜色&渐变动画/几何形状动画/WGGeometryAniViewController.m
//
//  WGGeometryAniViewController.m
//  WGOCAnimationPrimary
//
//  Created by wanggang on 2019/5/21.
//  Copyright © 2019 bozhong. All rights reserved.
//

#import "WGGeometryAniViewController.h"

@interface WGGeometryAniViewController ()

@property (nonatomic, strong) UIButton *btn;

@end

@implementation WGGeometryAniViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = UIColor.whiteColor;
    self.title = @"集合形状动画";
    
    self.btn = [[UIButton alloc] initWithFrame:CGRectMake(20, 100, SCREEN_WIDTH-20*2, 30)];
    [_btn setTitle:@"登录" forState:UIControlStateNormal];
    [_btn setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
    _btn.backgroundColor = [UIColor cyanColor];
    [self.view addSubview:_btn];
}

-(void)viewWillAppear:(BOOL)animated{
    
    [super viewWillAppear:animated];
    
    [UIView animateWithDuration:2 animations:^{
        
        self.btn.bounds = CGRectMake(0, 0, self.btn.width*0.8, self.btn.height*1.2);
    }];
}

@end

File: Swift/显示层初级动画/WGAnimationPrimary/WGAnimationPrimary/几何形状动画/WGGeometryAniViewController.swift
//
//  WGGeometryAniViewController.swift
//  WGAnimationPrimary
//
//  Created by wanggang on 2019/4/18.
//  Copyright © 2019 bozhong. All rights reserved.
//

import UIKit

class WGGeometryAniViewController: UIViewController {

    var loginBtn: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = UIColor.white
        
        loginBtn = UIButton(frame: CGRect(x: 20, y: 200, width: WgWith-20*2, height: 30))
        loginBtn.backgroundColor = UIColor.cyan
        loginBtn.setTitle("登录", for: .normal)
        view.addSubview(loginBtn)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        //动画开始
        UIView.beginAnimations(nil, context: nil)
        //动画持续时间
        UIView.setAnimationDuration(2)
        //最后状态
        loginBtn.bounds = CGRect(x: 0, y: 0, width: loginBtn.width*0.7, height: loginBtn.height*1.3)
        //提交动画
        UIView.commitAnimations()
    }
    
}
